schemas/groups/electrical.json

```diff
@@ -238,263 +252,428 @@
+                "$ref": "../definitions.json#/definitions/numberValue",
+                "units": "ratio"
+              },
+              "fieldDrive": {
+                "description": "% (0..100) of field voltage applied",
+                "$ref": "../definitions.json#/definitions/numberValue",
+                "units": "%"
+              },
+              "regulatorTemperature": {
+                "description": "Current temperature of critical regulator components",
+                "$ref": "../definitions.json#/definitions/numberValue",
+                "units": "K"
+              }
+            }
+         }
+      }
+    },
+    "solar": {
+      "description": "Data about Solar charging device(s)",
+      "patternProperties": {
+        "(^[A-Za-z0-9]+$)": {
+          "type": "object",
+          "title": "Solar",
+          "description": "Photovoltaic charging devices",
+          "allOf": [{
+            "$ref": "#/definitions/identity"
+          },{
+            "$ref": "#/definitions/dcQuantities"
+          },{
+            "$ref": "#/definitions/chargerQuantities"
+          }],
+          "properties": { 
+            "controllerMode": {
+              "type": "object",
+              "description": "The current state of the engine",
+              "allOf": [
+                {
+                  "$ref": "../definitions.json#/definitions/commonValueFields"
+                },
+                {
                   "properties": {
-                    "chargingAlgorithm": {
+                    "value": {
+                      "type": "string",
                       "enum": [
-                        "trickle",
-                        "two stage",
-                        "three stage",
-                        "constant current",
-                        "constant voltage",
-                        "custom profile"
+                         "off",
+                         "idle",
+                         "direct",
+                         "PWM",
+                         "MPPT"
                       ]
-                    },
-                    "chargerMode": {
+                    }
+                  }
+                }
+              ]
+            },
+            "panelVoltage": {
+                "description": "Voltage being supplied from Solar Panels to controller",
+                "$ref": "../definitions.json#/definitions/numberValue",
+                "units": "V"
+            },
+            "panelCurrent": {
+                "description": "Amperage being supplied from Solar Panels to controller",
+                "$ref": "../definitions.json#/definitions/numberValue",
+                "units": "A"
+            },
+            "panelTemperature": {
+                "description": "Temperature of panels",
+                "$ref": "../definitions.json#/definitions/numberValue",
+                "units": "K"
+            },
+
+            "load": {
```

Comment:
Should this be a boolean, something like `connected` or `underLoad`? If the value is n/a then just omit this key, otherwise it is true or false.